File: app/src/main/java/com/romellbolton/putmeon/database/TrackDao.java
package com.romellbolton.putmeon.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.romellbolton.putmeon.model.Track;

import java.util.List;

@Dao
public interface TrackDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveAll(List<Track> tracks);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void save(Track track);

    @Update
    void update(Track track);

    @Delete
    void delete(Track track);

    @Query("SELECT * FROM Track")
    LiveData<List<Track>> findAll();
}

File: app/src/main/java/com/romellbolton/putmeon/database/TrackDao.kt
package com.romellbolton.putmeon.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.romellbolton.putmeon.model.Track

@Dao
interface TrackDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveAll(tracks: List<Track?>?)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun save(track: Track?)

    @Update
    fun update(track: Track?)

    @Delete
    fun delete(track: Track?)

    @Query("SELECT * FROM Track")
    fun findAll(): LiveData<List<Track?>?>?
}